x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/optimizer/Optimizer.java

```diff
@@ -1278,8 +1284,11 @@ private Expression simplifyNot(Not n) {
                 return TRUE;
             }
 
-            if (c instanceof Negateable) {
-                return ((Negateable) c).negate();
+            if (c instanceof BinaryNegateable) {
```

Comment:
Case in point, it doesn't make any difference internally whether an expression is `BinaryNegtable` or `Unary` - the result in both cases is an `Expression` so why differentiate between the two.

Reply:
It was just because of the 2 different ifaces, but as you suggested we'd better merge them into one.